### DISCUSSION TITLE
Cannot upload more than input fields in supabase.

### DISCUSSION THREAD

**Discussion by RangerDevv** [supabase/supabase#12351] [Questions] [answered]
Hello! I am Ranger! I have recently started to learn supabase. I have run to this issue when trying to upload more than 2 content fields.  I am unable to see what the issue is. Help is appreciated.

`code: '23503', details: 'Key is not present in table "Documentation".', hint: null, message: 'insert or update on table "Contents" violates foreign key constraint "Contents_connect_fkey"'`

RLS policies: 
![image](https://user-images.githubusercontent.com/66392962/218270794-a86b2a50-b007-4b93-b4ad-ecd4d5468abb.png)

Location of the error:
Github Repo: https://github.com/RangerDevv/ReWrit/blob/main/src/components/svelte/newassignment.svelte 

Also I am using Astro, Tailwindcss, and Svelte.

**Comment by GaryAustin1** [comment, marked as answer]
Seems to be saying you have a foreign key relationship and are populating the fk in Contents table, but the value does not point to a row with that id in the Documents table.  Don't think RLS is involved.

**Reply by RangerDevv** [reply]
But I have a row that has the fk. I am not sure why it is saying that. Can you look at the code and tell me if I am doing something wrong?

**Reply by GaryAustin1** [reply]
If you are not getting errors back on the document insert then not sure (like RLS blocks document insert).  I can't tell what your fk is glancing at your code and without tables it is a guess.  Is the new document getting inserted?  How are you getting it's primary key to into your content insert?  Or is there some other column you have as fk?

**Reply by GaryAustin1** [reply]
Ah...
![image](https://user-images.githubusercontent.com/54564956/218272592-4f81ee94-43f2-4ecc-83c0-cfa283ade1d1.png)

Your content push is happening before your document write is finished.   You put it directly after an async call instead of in the then for the previous supabase call.   You either need to await the document insert or move your content insert into the then.

Not sure why you did not also use await for your first insert.  Your then seems just to be to log console.

**Reply by RangerDevv** [reply]
tysm!!
